dali/python/nvidia/dali/pipeline.py

```diff
@@ -246,15 +270,25 @@ def _prepare_graph(self):
                                 self._default_cuda_stream_priority)
         self._pipe.SetExecutionTypes(self._exec_pipelined, self._exec_separated, self._exec_async)
         self._pipe.SetQueueSizes(self._cpu_queue_size, self._gpu_queue_size)
-        prev_pipeline = Pipeline.set_current(self)
-        outputs = self.define_graph()
-        Pipeline.set_current(prev_pipeline)
+
+        if define_graph is not None:
+            if self._graph_out is not None:
+                raise RuntimeError("Duplicate graph definition - `define_graph` argument "
+                    "should not be specified when graph was define with a call to `set_outputs`.")
+        else:
+            define_graph = self.define_graph
+
+        with self:
+            if self._graph_out:
```

Comment:
What does assigning the `_graph_out` in `with self` context do? I understand that the call to `define_graph` has the current pipeline set, but this is just assignment, isn't it? It should already by created in the context earlier by the user.

Reply:
True.